Dart/src/com/jetbrains/lang/dart/Dart.bnf

```diff
@@ -251,14 +251,15 @@ classDefinition ::= metadata* 'abstract'? 'class' componentName typeParameters?
 mixinDeclaration ::= metadata* 'mixin' componentName typeParameters? onMixins? interfaces? classBody
 {mixin="com.jetbrains.lang.dart.psi.impl.AbstractDartPsiClass" implements="com.jetbrains.lang.dart.psi.DartClass"}
 
-enumDefinition ::= metadata* 'enum' componentName '{' enumConstantDeclaration (',' enumConstantDeclaration)* ','? '}'
-{pin=3  mixin="com.jetbrains.lang.dart.psi.impl.AbstractDartPsiClass" implements="com.jetbrains.lang.dart.psi.DartClass"}
+enumDefinition ::= metadata* 'enum' componentName typeParameters? mixins? interfaces? '{' enumConstantDeclaration? (',' enumConstantDeclaration)* ','? ';'? classMemberDefinition* '}'
+ {pin=3  mixin="com.jetbrains.lang.dart.psi.impl.AbstractDartPsiClass" implements="com.jetbrains.lang.dart.psi.DartClass"}
 
-enumConstantDeclaration ::= componentName
+enumConstantDeclaration ::= metadata* componentName typeArguments? arguments? |
+                            metadata* componentName typeArguments? '.' componentName arguments
```

Comment:
Any input matches `arguments?`, and this means that the second branch of this rule (after `|`) will never be reached. To make it work as expected, I flipped the branches. Test updated as well.
```
enumConstantDeclaration ::= metadata* componentName typeArguments? '.' componentName arguments |
                            metadata* componentName typeArguments? arguments?
```